site/zenodo_rdm/moderation/rules.py

```diff
@@ -123,3 +125,22 @@ def files_rule(identity, draft=None, record=None):
         score += current_scores.ham_files
 
     return score
+
+
+def match_query_rule(identity, draft=None, record=None):
+    """Calculate a score based on matched percolate queries against the given document in the specified index."""
+
+    document = record.dumps()
+    percolator_index = build_alias_name(f"moderation-queries-{record.index._name}")
+    matched_queries = current_search_client.search(
+        index=percolator_index,
+        body={"query": {"percolate": {"field": "query", "document": document}}},
```

Comment:
minor: also add a filter so that we only match `active: true` percolator queries